modules/spam/privatejokes.py

```diff
@@ -0,0 +1,108 @@
+import random
+
+
+from bs4 import BeautifulSoup
+from telegram import Update
+from telegram.ext import CallbackContext, CommandHandler, Filters, MessageHandler
+import requests
+
+
+from ..base import Base
+
+
+class PrivateJoke(Base):
+    def __init__(self, logger=None):
+        commandhandlers = [
+            CommandHandler("toutoutoutou", self.toutoutoutou),
+            CommandHandler("toutoutoutoum4a", self.toutoutoutoum4a),
+            CommandHandler(["whois", "whoissciper", "sciper"], self.whoissciper),
+            CommandHandler(["whoisnsfw", "whoisscipernsfw", "scipernsfw"], self.whoisscipernsfw),
+            CommandHandler(["genre", "gender", "sexe", "sex", "sexx", "genr"], self.gender),
+            MessageHandler(~Filters.command, self.carpe),
+            CommandHandler(["saisine", "ccg"], self.saisine),
+            CommandHandler(["horny"], self.horny),
+        ]
+        super().__init__(logger, commandhandlers, mediafolder="./media")
+
+    def toutoutoutou(self, update: Update, context: CallbackContext) -> None:
+        update.message.reply_sticker("CAACAgIAAxkBAAECenJg1f163I_8Uzc9UjymlOLV9yyxWAACywADwPsIAAEtUj0YdWOU7iAE")
+
+        self.logger.info("{} gets a toutoutoutou!".format(update.effective_user.first_name))
+
+    def toutoutoutoum4a(self, update: Update, context: CallbackContext) -> None:
+        update.message.reply_audio(audio=open(self._media("toutoutoutou.m4a"), "rb")).reply_sticker(
+            "CAACAgIAAxkBAAECenJg1f163I_8Uzc9UjymlOLV9yyxWAACywADwPsIAAEtUj0YdWOU7iAE"
+        )
+
+        self.logger.info("{} gets a toutoutoutou!".format(update.effective_user.first_name))
+
+    def whoissciper(self, update: Update, context: CallbackContext) -> None:
+        _, sciper = update.message.text.split(" ", 1)
+        try:
+            sciper = int(sciper)
+            if sciper < 100000 or sciper > 999999:
+                raise Exception  # epic coding
+        except:
+            update.message.reply_text("Not a valid SCIPER.")
+            return
+
+        URL = "https://people.epfl.ch/{}".format(sciper)
+        page = requests.get(URL)
+        soup = BeautifulSoup(page.content, "html.parser")
+
+        try:
+            name = soup.find_all("h1", class_="mr-3")[0].text
+        except:
+            update.message.reply_text("SCIPER not found or not public.")
+            return
+
+        update.message.reply_text("{} is {}.".format(sciper, name))
+
+    def whoisscipernsfw(self, update: Update, context: CallbackContext) -> None:
+        _, sciper = update.message.text.split(" ", 1)
+        try:
+            sciper = int(sciper)
+            if sciper < 100000 or sciper > 999999:
+                raise Exception  # epic coding

```

Comment:
Instead, raise ValueError (or custom exception), and catch precisely that below. Won't catch "unexpected exceptions"